### DISCUSSION TITLE
Type annotations for `reactive(monotonic)` in the tutorial?

### DISCUSSION THREAD

**Discussion by lauralindzey** [Textualize/textual#3984] [Q&A] [answered]
I needed a timer in the app I'm building, so I was delighted that your [tutorial](https://textual.textualize.io/tutorial/#stopwatch-application) includes one. 

However, I'm stuck on what the proper type annotation is for `start_time = reactive(monotonic)`. 
I directly copied my implementation from stopwatch05.py, about which mypy also complains:
``` 
$ mypy stopwatch05.py 
stopwatch05.py:12: error: Need type annotation for "start_time"  [var-annotated]
stopwatch05.py:12: error: Argument 1 to "reactive" has incompatible type "Callable[[], float]"; expected "Callable[[], <nothing>]"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```

**Comment by davep** [comment, marked as answer]
The proper type for `start_time` would be `reactive[float]`:

```diff
diff --git a/docs/examples/tutorial/stopwatch05.py b/docs/examples/tutorial/stopwatch05.py
index 19f6366f7..66a2162a2 100644
--- a/docs/examples/tutorial/stopwatch05.py
+++ b/docs/examples/tutorial/stopwatch05.py
@@ -9,7 +9,7 @@ from textual.widgets import Button, Footer, Header, Static
 class TimeDisplay(Static):
     """A widget to display elapsed time."""
 
-    start_time = reactive(monotonic)
+    start_time: reactive[float] = reactive(monotonic)
     time = reactive(0.0)
 
     def on_mount(self) -> None:
```

The reason is `start_time` is using `monotonic` as [a dynamic default](https://textual.textualize.io/guide/reactivity/#dynamic-defaults) and [`monotinic` returns a `float`](https://docs.python.org/3/library/time.html#time.monotonic). This also explains why mypy will complain about `start_time` but not `time` in the example, as it can easily infer the type of `time`.
